### PR TITLE
replace span_lint with span_lint_and_sugg along with error message

### PR DESCRIPTION
**Pull request by iobtl**
fixes: #6874 
changelog: none

apologies if this may not be the most idiomatic way of doing it, any advice on changes (if any) would be greatly appreciated.

### CODE REVIEW

File: tests/ui/unnecessary_cast.stderr [hunk 1]
```diff
@@ -2,21 +2,21 @@ error: casting to the same type is unnecessary (`i32` -> `i32`)
   --> $DIR/unnecessary_cast.rs:6:5
    |
 LL |     1i32 as i32;
-   |     ^^^^^^^^^^^
+   |     ^^^^^^^^^^^ help: try: `1i32`
```
Comment (llogiq) [inline tests/ui/unnecessary_cast.stderr:5]: Wouldn't we usually suggest `1_i32` for better legibility?

**Comment by llogiq** [comment]
Thank you for this improvement. I just have one question at the moment, otherwise this seems fine.

**Comment by iobtl** [comment]
I see, that makes sense indeed! Thank you for the suggestion, let me make and push the changes to adhere to that standard

File: tests/ui/unnecessary_cast.stderr (hunk 1 above)
Comment (camsteffen) [reply tests/ui/unnecessary_cast.stderr:5]: See #6232

**Comment by llogiq** [comment]
Thank you! @bors r+

**Pull request merged**